kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -1,341 +1,465 @@
+		try {
+	
+        	 String bucketName=null;
+        	 String finalObjectName=null;
+        	if(useAccountAsBucketname) {
+        		 bucketName=account;
+        		 finalObjectName = ObjectStoreUtil.getName(container,null,TAGS_FILENAME);
+        	}else {
+        		 bucketName=container;
+        		 finalObjectName = TAGS_FILENAME;
+        	}
+			AmazonS3 connection = getConnection(bucketName);
+			if (!connection.doesBucketExistV2(bucketName))
+	            connection.createBucket(bucketName);
+			
+			Map<String, String> existingMetadata=getTags(account, container);
+			if(!connection.doesObjectExist(bucketName, finalObjectName)) {
+				connection.putObject(bucketName, finalObjectName, "");
+			}
+			
+			tags.entrySet().stream()
+					.forEach(m -> existingMetadata.put(m.getKey(), m.getValue() != null ? m.getValue() : null));
+			List<Tag> tagSet=new ArrayList<Tag>();
+			for(Entry<String, String> entry:existingMetadata.entrySet()) {
+				tagSet.add(new Tag(entry.getKey(), entry.getValue()));
+			}
+			ObjectTagging objectTagging=new ObjectTagging(tagSet);
+			
+			SetObjectTaggingRequest setObjectTaggingRequest=new SetObjectTaggingRequest(bucketName,finalObjectName,objectTagging);
+			connection.setObjectTagging(setObjectTaggingRequest);
+			
+
+		} catch (Exception e) {
+			LOGGER.error(SESSIONID, REGISTRATIONID, "Exception occured while addTags for : " + container,
+					ExceptionUtils.getStackTrace(e));
+			throw new ObjectStoreAdapterException(OBJECT_STORE_NOT_ACCESSIBLE.getErrorCode(),
+					OBJECT_STORE_NOT_ACCESSIBLE.getErrorMessage(), e);
+		}
+		return tags;
+	}
+
+	@Override
+	public Map<String, String> getTags(String account, String container) {
+		Map<String, String> objectTags = new HashMap<String, String>();
+		try {
+	
+		 String bucketName=null;
+		 String finalObjectName=null;
+     	if(useAccountAsBucketname) {
+     		 bucketName=account;
+     		 finalObjectName = ObjectStoreUtil.getName(container,null,TAGS_FILENAME);
+     	}else {
+     		 bucketName=container;
+     		 finalObjectName = TAGS_FILENAME;
+     	}
+		AmazonS3 connection = getConnection(bucketName);
+		if(connection.doesObjectExist(bucketName, finalObjectName)) {
+		
+		GetObjectTaggingRequest getObjectTaggingRequest=new GetObjectTaggingRequest(bucketName,finalObjectName);
+		
+			GetObjectTaggingResult getObjectTaggingResult=connection.getObjectTagging(getObjectTaggingRequest);
+		
+		if(getObjectTaggingResult!=null) {
+			List<Tag> tagSet = getObjectTaggingResult.getTagSet();
+			if (tagSet != null) {
+				for(Tag tag:tagSet) {
+					objectTags.put(tag.getKey(), tag.getValue());
+				}
+			}
+		  }
+		}
+		}catch(Exception e){
+			LOGGER.error(SESSIONID, REGISTRATIONID, "Exception occured while getTags for : " + container,
+					ExceptionUtils.getStackTrace(e));
+			throw new ObjectStoreAdapterException(OBJECT_STORE_NOT_ACCESSIBLE.getErrorCode(),
+					OBJECT_STORE_NOT_ACCESSIBLE.getErrorMessage(), e);
+		}
+		return objectTags;

```

Comment:
If we throwning the above exception, then can we just return inside the try block itself ?